modules/contextToProps.tsx

```diff
@@ -1,19 +1,39 @@
 import * as React from 'react'
 
 export default function contextToProps<P, C>(
-    Consumer: React.ComponentType<React.ConsumerProps<C>>
+    Consumer: React.ComponentType<React.ConsumerProps<C>>,
+    config?: string | ((props: P) => C)
```

Comment:
This should be `C => P`

Reply:
Actually it should be C => 3rd generic type. Currently `BaseComponent` is `P & C`, a 3rd generic type should be passed along (default to P & C).